### ISSUE TITLE
Upgrade to rxjava 3.x

### ISSUE THREAD

**Issue by ray-roestenburg-da** [digital-asset/daml#15180]

**Comment by S11001001** [comment]
We should exclude any parts of `bindings-rxjava` that are currently deprecated from `bindings-rxjava3` at the time of this copy.

**Comment by S11001001** [comment]
I've come around to @stefanobaghino-da 's idea that we ought to clean up the overloads with builders or other argument-set representations before doing the rxjava3 split, and then delete the old deprecated functions in rxjava3 so we have a clean base.

**Comment by stefanobaghino-da** [comment]
> I've come around to @stefanobaghino-da 's idea that we ought to clean up the overloads with builders or other argument-set representations before doing the rxjava3 split, and then delete the old deprecated functions in rxjava3 so we have a clean base.

Any specific reason why you'd favor my take?

**Comment by S11001001** [comment]
@stefanobaghino-da So that we have a good stepped approach to simply not having the deprecated functions in rxjava3.

**Comment by stefanobaghino-da** [comment]
> @stefanobaghino-da So that we have a good stepped approach to simply not having the deprecated functions in rxjava3.

Makes sense. I think it's cleaner to start with a library free of deprecations, as long as we can have a migration path.

**Comment by stefanobaghino-da** [comment]
I would also recommend evaluating whether something can be done for the overloads with a security token. I think a gRPC interceptor might be part of the solution.

**Comment by ray-roestenburg-da** [comment]
Sounds good
